framework/src/tinyalsa/tinyalsa.c

```diff
@@ -338,17 +333,14 @@ static int pcm_set_config(struct pcm *pcm, const struct pcm_config *config)
 	}
 
 	/* Gain buffer size and count through the ioctl */
-	buf_info.buffer_size = config->period_size;
```

Comment:
config->period_size & config->period_count is basic of tinyalsa usage, we should keep.
Also, This code means below flow.
1. user set buffer size & count
2. Ask to driver that it is acceptable
3. driver check boundary of buffer size & count adjust buffer info if it needed.
4. If driver doesn't define ioctl regarding AUDIOIOC_GETBUFFERINFO, use default configuration value.
So I think most of values in pcm structure should be kept.

Reply:
Ok. I will keep these variables. 
But in the driver code, I dont understand why we are checking like below?

buf_size = bufinfo->buffer_size * (priv->bpsamp >> 3) * priv->nchannels;
                if (buf_size > DMA_BUFFER_MAX_SIZE || buf_size < DMA_BUFFER_MIN_SIZE) {
                        bufinfo->buffer_size = CONFIG_ALC5658_BUFFER_SIZE;
                }

The buffer that we allocate and send to DMA will be bufinfo->buffer_size. So, maybe we need to change this in driver code.

Reply:
However, I have removed some other fields from the pcm and the config structures, since they are not being used. If required in future for any new code, then we can add them. Is that OK?

Reply:
first question : yes it will be changed based on inflight number.
second : sure why not?

Reply:
Regarding my first question, this is what i mean.

for example buffer_size = CONFIG_ALC5658_BUFFER_SIZE (which is 2048 bytes)
So, each buffer in our queue will contain max 2048 bytes of data (irrespective of bpsamp and nchannels) 
For example, if bpsamp = 16 bits and nchannels = 2, then buffer size is not 2048 x 2 x 2. 
It is always 2048 bytes. So i feel we must be checking bufinfo->buffer_size only. 

Anyway, it is not relevant to this commit !!! So, sorry for bringing it up here !!